Track.h

```diff
@@ -109,12 +127,20 @@ class Track
 private:
   TrackState state_;
   HitVec hits_;
-  HitVec initHits_;
-  HitIdxVec hitIdxVec_;
   float chi2_;
+  unsigned int mcTrackID_;
+  unsigned int nHitsMatched_;
+  unsigned int seedID_;
+  unsigned int duplicateID_;
+  bool isDuplicate_;
+  HitIdxVec hitIdxVec_;
```

Comment:
@cerati @dan131riley @osschar 

I am a bit stuck here for a number of reasons.  I am now remembering why I decided to explicitly make some ints change to unsigned ints.
If you look above, you will see I changed layer() to unsigned, to match what Geometry.h/.cc yields for the layer info.  

However, the eta binning structures between SMatrix and Matriplex use eta bins as either ints or unsigned.  Since we would rather port the overlap to SMatrix, it would make sense to move the bins in Smatrix to ints and not unsigned.

I have a working version of the smatrix code changed appropriately for ints.  THis meant changing the Geometry code to ints, some things in Propagation, etc.  Math/ was left unchanged.  Additionally, anything that does not have an explicit comparison with my changes was left as is (some unsigned ints remaining).  

I checked the validation of the smatrix, and the results of changes to many ints to my version is identical.  

under: ttree-merge-2-int